### PR TITLE
Issue 7 server side rendering

### LINKED ISSUE #7: Server Side Rendering
You can add SSR to this or can take my commits in the `feature/server-side-rendering` branch and fix it.

The initial launch programs landing page ideally should be server-side-rendered.
A boilerplate to implement the Server-side rendering can be used.

### PR DESCRIPTION
**Pull request by maelannb**
Resolve #7 

I added the server side rendering thanks to `ReactDOMServer`, I also modified the `index.js` of React to get the possibility to hydrate the dom if needed.

I took the liberty of modifying the development server because the server side rendering depends of the react app being built.
To start the development server it's now advised to use `npm run dev`.

I hope this solution is what you wanted 🚀

### CODE REVIEW

**Comment by jvsinghk** [comment]
@setheal Can you help me understand your approach? By the steps, you followed to add SSR to it. Please mention the resource or documentation used if any. Which dependencies need to be installed in what fashion to achieve the same result. Please resolve the conflicts.

**Review by jvsinghk — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts.

File: package.json [hunk 1]
```diff
@@ -3,17 +3,23 @@
...
+    "ignore-styles": "^5.0.1",
     "react": "^16.13.1",
     "react-bootstrap": "^1.3.0",
     "react-dom": "^16.13.1",
     "react-scripts": "3.4.1"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "react-scripts build && node ./server",
+    "dev": "nodemon --ignore 'build/*' --exec \"yarn start\"",
```
Comment (jvsinghk) [inline package.json]: After running `npm run dev` the yarn start is starting and restarting again and again in the loop. Please install the latest dependencies check on your local update me. I couldn't test it using `npm run dev` after installing the dependencies. I have some changes in the `package.json` and `package-lock.json`

**Review by jvsinghk — COMMENTED** [review]
Couldn't run it using `npm run dev` the `yarn start` in running in the loop

File: package.json (hunk 1 above)
Comment (maelannb) [reply package.json]: Hey @jvsinghk I will take a look as soon as possible :)

Comment (jvsinghk) [reply package.json]: Hey, @setheal Sure thanks :)

Comment (jvsinghk) [reply package.json]: Hey, @setheal any updates?

**Comment by setheal** [comment]
Hey @jvsinghk, I checked on my side for your issue with `npm run dev` and I can't replicate it. Can you post your node version please ?

File: package.json (hunk 1 above)
Comment (maelannb) [reply package.json]: Hey @jvsinghk, I checked on my side for your issue with `npm run dev` and I can't replicate it. Can you post your node version please ?

Comment (jvsinghk) [reply package.json]: node -v `12.19.0` , npm -v `6.14.8` and yarn `1.22.5`. Let me know if I need to update or fix anything. I am updating dependencies from my side in #15 PR. Once it gets merged, please fetch master.

Comment (maelannb) [reply package.json]: Okay, I'll wait for you and try with your version :)

Comment (jvsinghk) [reply package.json:22]: I have updated and merged the changes. Please don't use `npm` from now on. Just run `yarn` and `yarn start` after fetching master and deleting `node_modules`. I have removed `package-lock.json` hope it doesn't make any breaking changes.

Comment (jvsinghk) [reply package.json:22]: Merged pr, updated node to `12.19.0`. Can you post your node version? I hope now it will work.

Comment (jvsinghk) [reply package.json]: Hey @setheal, it's done :)

**Comment by jvsinghk** [comment]
@setheal I'm sorry, I have accidentally closed it. Do

**Comment by jvsinghk** [comment]
Screenshot (1597)
Screenshot (1598)
Hey, @setheal 'yarn start` works perfectly for SSR 
`npm run dev` which interns run `yarn start` runs into an infinite loop. Kindly, do the necessary changes.

**Comment by setheal** [comment]
> Screenshot (1597)
> Screenshot (1598)
> Screenshot (1599)
> Hey, @setheal `yarn start` works perfectly for SSR
> `npm run dev` which interns run `yarn start` runs into an infinite loop. Kindly, do the necessary changes.
> I have actually removed `package-lock.json` #15 to remove `npm` dependencies but when I saw that this PR has changes for it. Added that back in master. I was thinking of keeping `yarn-lock.json` only. Let me know what you think?

Hey @jvsinghk, sory for the late answer I'm working on my thesis rigth now, I should be available thursday to focus on this issue.
To answer your question, it's always better to use only one package manager, so if you want to work only with Yarn, go ahead. :)

**Comment by maelannb** [comment]
Hey @jvsinghk I looked on your screenshot and it seems like nodemon is not ignoring the build folder that's why it keep restarting. Can you check that you have this part in the `dev` script : `--ignore 'build/*'` ? Thank you.

EDIT: nevermind I just saw it in your screenshot... Can you run a `npm run build` and check if another files/folder is modified by the command ?

**Comment by jvsinghk** [comment]
Hey, @setheal I do have that. Before running `npm run build` I had some changes in `package-lock.json` and `yarn.lock` which I got after running `npm install` and `yarn` respectively. After running `npm run build` the build folder gets updated. and then I ran `npm run dev` which runs into the same issue. I have committed my changes if you allow me I can push them to your branch. That might help you to reproduce the bug.

**Comment by maelannb** [comment]
@jvsinghk yes, push them pelase

**Comment by jvsinghk** [comment]
@setheal pushed it. review it, please

**Comment by maelannb** [comment]
Hye @jvsinghk everythings works as a charm on my side. Are you still encountering this issue ?

**Comment by jvsinghk** [comment]
> Hye @jvsinghk everything works like a charm on my side. Are you still encountering this issue?

Hey @setheal Yes, check maybe you haven't pushed any part of the code either it's  in .`gitignore` or something else
`npm run dev

> spacex@0.1.0 dev G:\Open-Source-Review\setheal\spacex
> nodemon --ignore 'build/*' --exec "yarn start"

[nodemon] 2.0.5
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `yarn start`
yarn run v1.22.5
$ react-scripts build && node ./server
yarn run v1.22.5mized production build...
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
yarn run v1.22.5ng `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
yarn run v1.22.5
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
yarn run v1.22.5
$ react-scripts build && node ./server`

`>yarn run dev
yarn run v1.22.5
$ nodemon --ignore 'build/*' --exec "yarn start"
[nodemon] 2.0.5
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
Creating an optimized production build...
[nodemon] restarting due to changes...
[nodemon] starting `yarn start`
$ react-scripts build && node ./server
`

**Comment by maelannb** [comment]
Everything works as expected on my side, I'm really confused. Which OS are you running ?

**Comment by jvsinghk** [comment]
> Everything works as expected on my side, I'm really confused. Which OS are you running?

I don't think the issue is related to OS. As on my other projects, the dev script runs like a charm. It's Windows btw.
If you running `yarn start` it works but `yarn run dev` runs into a loop.

**Comment by jvsinghk** [comment]
@AssaultKoder Please review it

**Comment by jvsinghk** [comment]
@AssaultKoder95 Please review it.

**Comment by maelannb** [comment]
hey @jvsinghk can you check the latest version please ?

**Comment by jvsinghk** [comment]
> hey @jvsinghk can you check the latest version, please?

hey, @setheal checked the latest version, doesn't work.

**Pull request closed without merging**